### ISSUE TITLE
[RafteryCopula] broken tests

### ISSUE THREAD

**Issue by lrnv** [lrnv/Copulas.jl#98]
@Santymax98 There are broken tests in Rafterycopula, as when we merged your PR we forgot to run them...

[On the pdf here](https://github.com/lrnv/Copulas.jl/blob/c9046d22ccdffdf5649cdb756d10abf7a0811128/test/RafteryTest.jl#L25-L26)

[On the cdf there](https://github.com/lrnv/Copulas.jl/blob/c9046d22ccdffdf5649cdb756d10abf7a0811128/test/RafteryTest.jl#L49)

This is problematic, I had the pdf retuninrg negative values on some test cases. Would you mind checking again the coirrectness of your code ? 

Cheers

**Comment by Santymax98** [comment]
Maybe it was a typing error, I will check the manual calculations again.

I'll start reviewing them in a moment.

**Comment by lrnv** [comment]
You should be able to extract the d value from `length(C)` usually. You think that we missed one in the package's code ?

**Comment by lrnv** [comment]
Maybe 

https://github.com/lrnv/Copulas.jl/blob/bbfd73bcccbf8b58b745bba91df60361c4c68fcc/src/MiscellaneousCopulas/RafteryCopula.jl#L83C5-L83C20

should be `d` and not `dim` ? But this is in the random generator and not in the pdf / cdf so not the same thing

**Comment by Santymax98** [comment]
It is very strange, because the code independently is giving the correct values. As for `dim`, it is not causing a problem in Rand, because it depends on the dimension of the entered vector. 

In my code R[1] corresponds to the parameter and R[2] corresponds to the dimension.

**Comment by Santymax98** [comment]
Yes, there is an error because it is giving me the value of `2` as a dimension and it should be 3, however when I substitute 3 in that line. I get the correct value. 
In general, it is correct and the value is the same as what I obtained by calculating it by hand. I think you should try to access the value of `d` and ensure that it is the dimension of the copula. I think I could fix it by using `dim = length(x)`

**Comment by lrnv** [comment]
So you are saying that the values written in the tests are wrong, and that the value returned by the package is the right one, so that we should just update the test ?

**Comment by lrnv** [comment]
Hum... This is weird indeed. I just check by eyeball that the package code is OK with respect to the formula tfor the CDF and the PDF in the paper, and it looks OK to me (except the fact that `log`-ed values in the pdf looks negative to me, but the cdf looks ok).

**Comment by lrnv** [comment]
In the pdf computation, you may change :
```julia
    logpdf_value = log(term_numerator) - log(term_denominator) + log(term_product)
```

for : 

```julia
    logpdf_value = log(-term_numerator) - log(-term_denominator) + log(term_product)
```

because `term_denominator = (1 - R.θ)^(d - 1) * (1 - R.θ - d)` is clearly negative since theta is in [0,1] and d is minimum 2, and `term_numerator = 1 - d - R.θ * u_ordered[d]^((1 - R.θ - d) / (1 - R.θ))` also looks negative.
